addons/website_event_track/controllers/event_track.py

```diff
@@ -319,34 +319,33 @@ def _split_track_by_days(self, track, local_tz):
         Also return a set of all the time slots
         """
         start_date = fields.Datetime.from_string(track.date).replace(tzinfo=pytz.utc).astimezone(local_tz)
-        start_datetime = self.time_slot_rounder(start_date, 15)
-        end_datetime = self.time_slot_rounder(start_datetime + timedelta(hours=(track.duration or 0.25)), 15)
-        time_slots_count = int(((end_datetime - start_datetime).total_seconds() / 3600) * 4)
+        track_start_time_slot = start_time_slot = self.time_slot_rounder(start_date, 15)
+        end_datetime = self.time_slot_rounder(start_time_slot + timedelta(hours=(track.duration or 0.25)), 15)
+        time_slots_count = int(((end_datetime - start_time_slot).total_seconds() / 3600) * 4)
 
-        time_slots_by_day_start_time = {start_datetime: 0}
+        time_slots_by_day_start_time = {start_time_slot: 0}
+        next_day = (start_time_slot + timedelta(days=1)).date()
         for i in range(0, time_slots_count):
             # If the new time slot is still on the current day
-            next_day = (start_datetime + timedelta(days=1)).date()
-            if (start_datetime + timedelta(minutes=15*i)).date() <= next_day:
-                time_slots_by_day_start_time[start_datetime] += 1
+            time_slots = track_start_time_slot + timedelta(minutes=15 * i)
```

Comment:
Naming should make sense, this isn't "slots", it's a datetime.
Also why multiply when we're already adding slots of 15 minutes every loop anyway?
```suggestion
            current_slot_end = current_slot_start + timedelta(minutes=15)
```